libsolidity/interface/StandardCompiler.cpp

```diff
@@ -180,9 +212,19 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 				ReadFile::Result result = m_readFile(url.asString());
 				if (result.success)
 				{
-					m_compilerStack.addSource(sourceName, result.contentsOrErrorMessage);
-					found = true;
-					break;
+					if (!hash.empty() && !hashMatchesContent(hash, result.contentsOrErrorMessage))
+						errors.append(formatError(
+							false,
+							"IOError",
+							"general",
+							"Mismatch between content and supplied hash for \"" + url.asString() + "\""
```

Comment:
Can you provide the file name and the url? The url might just be a cryptic swarm hash.